pkg/apiserver/authenticator/authn.go

```diff
@@ -58,30 +59,37 @@ type AuthenticatorConfig struct {
 
 // New returns an authenticator.Request or an error that supports the standard
 // Kubernetes authentication mechanisms.
-func New(config AuthenticatorConfig) (authenticator.Request, error) {
+func New(config AuthenticatorConfig) (authenticator.Request, *common.SecurityDefinitions, error) {
 	var authenticators []authenticator.Request
+	securityDefinitions := common.NewSecurityDefinition()
 
 	// BasicAuth methods, local first, then remote
 	if len(config.BasicAuthFile) > 0 {
 		basicAuth, err := newAuthenticatorFromBasicAuthFile(config.BasicAuthFile)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		authenticators = append(authenticators, basicAuth)
+		if err := securityDefinitions.AddBasicDefinition("HTTP"); err != nil {
+			return nil, nil, err
+		}
 	}
 	if len(config.KeystoneURL) > 0 {
 		keystoneAuth, err := newAuthenticatorFromKeystoneURL(config.KeystoneURL)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		authenticators = append(authenticators, keystoneAuth)
+		if err := securityDefinitions.AddBasicDefinition("Keystone"); err != nil {
```

Comment:
For this and duplicate bearer definitions, I was thinking this could have a documentation/discovery benefit as users can look at it and see what kind of authentication supported by this server. you are absolutely right that without more information it has less uses and clients are not using these duplicates at all and thread them the same. I would remove them for now.